src/main/java/com/exactpro/th2/crawler/util/CrawlerUtils.java

```diff
@@ -185,6 +185,31 @@ public static Interval updateMessageRecoveryState(IntervalsWorker worker, Interv
         return data;
     }
 
+    public static RecoveryState.InnerMessage findPairedMessage(RecoveryState.InnerMessage message,
+                                                               Map<AliasAndDirection, RecoveryState.InnerMessage> messages) {
+        Direction direction = Direction.valueOf(message.getDirection().toString());
```

Comment:
Why can you simply use the direction from the message?

Reply:
Becauese the message has another Direction (the one from Cradle, not from gRPC)